Recipes/InheritedTypeJsonConverter/InheritedTypeJsonConverter.cs

```diff
@@ -239,5 +194,69 @@ private IEnumerable<Type> GetChildTypes(Type type)
 
             return allChildTypes[type];
         }
+
+        private IEnumerable<CandidateChildType> GetCandidateChildTypes(Type objectType, HashSet<string> jsonProperties)
+        {
+            var childTypes = GetChildTypes(objectType);
```

Comment:
Technically it could still be an extension method but we just don't like the trade offs here to get that.